cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/adapter/AdapterServiceTests.java

```diff
@@ -98,14 +100,36 @@ public void testAdapters() throws Exception {
     Assert.assertTrue(adapters.isEmpty());
   }
 
+  @Test
+  public void testInvalidJars() throws Exception {
+    Class<?> clz = AdapterApp.class;
+    String adapterType = "adapterType";
 
-  // TODO: Negative tests for deploying adapters
+    Attributes attributes = generateAttributes(clz, adapterType);
+    setupAdapterJarWithManifestAttributes(clz, attributes);
 
-  private static void setupAdapters() throws IOException {
-    setupAdapter(AdapterApp.class, "dummyAdapter");
+    // Using a valid manifest results in the adapterTypeInfo being registered
+    adapterService.registerAdapters();
+    Assert.assertNotNull(adapterService.getAdapterTypeInfo(adapterType));
+
+
+    List<String> requiredAttributes = ImmutableList.of("CDAP-Source-Type", "CDAP-Sink-Type",
+                                                       "CDAP-Adapter-Type", "CDAP-Adapter-Program-Type");
+
+    // removing the any of the required attributes from the manifest results in the AdapterTypeInfo not being created.
+    for (int i = 0; i < requiredAttributes.size(); i++) {
+      adapterType = String.format("%s%s", adapterType, i);
+      attributes = generateAttributes(clz, adapterType);
+      Object removed = attributes.remove(new Attributes.Name(requiredAttributes.get(i)));
```

Comment:
Removing a required parameters is a bit confusing in the test. Add another function
```generateAttributes(Class<?> clz, String AdapterType, Map<String, String> attributes)```
and from 
```generateAttributes(Class<?> clz, String adapterType)```
call the function with the additional parameter